### ISSUE TITLE
RFE: check for label capitalization

### ISSUE THREAD

**Issue by scop** [go-critic/go-critic#1209]
A check for capitalization of labels would be a nice addition. It seems starting with uppercase rather than lowercase letter is more conventional, so that's what I'd like the check to demand. Not sure if this counts as opinionated.
- Example in Effective Go: https://go.dev/doc/effective_go#switch
- Example in Go language spec: https://go.dev/ref/spec#Labeled_statements
- Somewhat more common in the Go codebase according to some greps on both possible labels (hits a bunch of false positives) and label usages (very few false positives, but multiple hits per label skews results)

I can have a look at implementing this if it's something we'd like to have.

**Comment by cristaloleg** [comment]
I like it! Not sure if this can be easily solved with ruleguard, @quasilyte ? :)

**Comment by quasilyte** [comment]
https://github.com/quasilyte/go-consistent infers your source code convention and enforces it.
See https://github.com/quasilyte/go-consistent#label-case

We could add a checker if there is a common convention. That would require checking any big Go corpus.
Like this one: https://quasilyte.dev/gocorpus/
Queries like this can find labeled statements:
```
$label: for $*_ { $*_ }
```
```
$label: if $*_ { $*_ }
```
```
$label: switch $*_ { $*_ }
```
Etc.

**Comment by scop** [comment]
I ran the above plus `$label: return $*_` against all repos, results:

```shellsession
$ wc -l labels.txt 
204 labels.txt
$ grep -c '^[A-Z][a-z]' labels.txt
62
$ grep -Ec '^[^a-z]+$' labels.txt 
20
$ grep -c '^[a-z]' labels.txt 
122
```
The data: [labels.txt](https://github.com/go-critic/go-critic/files/8355211/labels.txt)

(I would have liked to run it against "any label, except `default:` within a switch", or just "any label" + grepping `default:` out of the results, but couldn't figure out how to do it, and clicking the help button didn't do anything.)

Anyway, with this test set, roughly 60% is camelcase starting with lowercase (or all lowercase), 30% camelcase starting with uppercase, and 10% all uppercase. Despite the result, I would still personally go with camelcase starting with uppercase because of the reasons from the initial comment here, especially language spec and Effective Go.
